includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```
@@ -2007,6 +2016,12 @@ public function process_order_for_confirmed_intent( $order, $intent_id, $save_pa
 
 		if ( ! empty( $error ) ) {
 			WC_Stripe_Logger::error( 'Error when processing payment for order: ' . $order->get_id(), [ 'error_message' => $error->message ] );
+			// If the intent is back to requires_payment_method, the customer cancelled the redirect
+			// (e.g. closed the Klarna popup). Treat this as a recoverable cancellation so the
+			// order is not permanently failed and the shopper can retry with another method.
+			if ( isset( $intent->status ) && WC_Stripe_Intent_Status::REQUIRES_PAYMENT_METHOD === $intent->status ) {
+				throw new WC_Stripe_Payment_Cancelled_Exception( $error->message );
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '2023', column '15'). (undefined)

(MissingImport)